### ISSUE TITLE
[strict-boolean-expressions] Allow guaranteed non empty strings

### ISSUE THREAD

**Issue by niklasf** [typescript-eslint/typescript-eslint#1832]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": ["error", { "allowSafe": true, "allowNullable": true }]
  }
}
```

```TS
type Union = 'a' | 'b';

function f(): Union | undefined {
  return 'a';
}

if (f()) { // <-- line 7
  // ...
}
```

**Expected Result**

Passes linting.

**Actual Result**

```
  7:5  error  Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly  @typescript-eslint/strict-boolean-expressions
```

**Additional Info**

This might be considered a feature request, not a bug.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.26.0` |
| `@typescript-eslint/parser`        | `2.26.0` |
| `TypeScript`                       | `2.2.2` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `v13.12.0` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
Duplicate of #1785
